### ISSUE TITLE
Use PyStructSequence for result() return value

### ISSUE THREAD

**Issue by encukou** [python-ldap/python-ldap#61]
The [result](http://python-ldap.readthedocs.io/en/latest/reference/ldap.html#ldap.LDAPObject.result)/[result2](http://python-ldap.readthedocs.io/en/latest/reference/ldap.html#ldap.LDAPObject.result2)/[result3](http://python-ldap.readthedocs.io/en/latest/reference/ldap.html#ldap.LDAPObject.result3)/[result4](http://python-ldap.readthedocs.io/en/latest/reference/ldap.html#ldap.LDAPObject.result4) API looks quite silly by today's standards, if you don't mind me saying.

Do what [os.stat does](https://docs.python.org/3/library/os.html#os.stat_result). `result()` should return an object that:
- for backwards compatibility, acts like a tuple of the first two elements
- has everything accessible as attributes.

Such a type can be defined at the C level using [Struct Sequence](https://docs.python.org/3/c-api/tuple.html#struct-sequence-objects).
In Python 2 this API is private; use a regular namedtuple subclass?

Deprecate result2/result3/result4 (but keep them around indefinitely).

**Comment by spaceone** [comment]
result4 is definetly used in the real world because it's the only way to get the server control response.

**Comment by encukou** [comment]
Indeed, and `result4()` should stay as it is. The proposed usage for `result()` is something like:

```
result = connection.result()
use(result.type)
use(result.data)
use(result.msgid)
use(result.controls)
use(result.name)
use(result.value)
```

with `type, data = connection.result()` still working, for backwards compatibility

**Comment by spaceone** [comment]
This would be very nice.
